src/robusta/core/discovery/discovery.py

```diff
@@ -54,9 +61,49 @@ class Config:
         arbitrary_types_allowed = True
 
 
+DISCOVERY_PIPE = "/tmp/discovery_pipe"
+SHUTDOWN_SIGNAL = "shutdown"
+STACKDUMP_SIGNAL = "stackdump"
+
+
 class Discovery:
     executor = ProcessPoolExecutor(max_workers=1)  # always 1 discovery process
 
+    @staticmethod
+    def init_discovery_error_pipes():
+        try:
+            os.mkfifo(DISCOVERY_PIPE)
+        except FileExistsError:
+            pass
+        except Exception:
+            logging.error(
+                "Failed to create discovery pipes",
+                exc_info=True,
+            )
+
+    @staticmethod
+    def send_signal_to_pipe(signal):
+        try:
+            with open(DISCOVERY_PIPE, mode="w") as pipe:
```

Comment:
double check that this can never be blocking, (I had issues with other types of pipes i opened)

Reply:
in practice if the pipe wasnt open there was no issue, but check in docs

Reply:
@Avi-Robusta what do you mean by blocking?
We don't want it to effect the discovery process on any occasion